lib/child_process.js

```diff
@@ -473,7 +473,7 @@ function spawnSync(/*file, args, options*/) {
 
   // Validate maxBuffer, if present.
   if (options.maxBuffer != null &&
-      !(Number.isInteger(options.maxBuffer) && options.maxBuffer >= 0)) {
+      !(typeof options.maxBuffer === 'number' && options.maxBuffer >= 0)) {
     throw new TypeError('"maxBuffer" must be an unsigned integer');
```

Comment:
0.1 is technically fine as a `maxBuffer` value, even if it is a bit weird. We're just going to be doing comparisons against the value to see if we surpass it.

> And the error message should certainly be updated but that would upgrade this to a semver-major change

This builds on an existing semver major change, so this is going to be semver major regardless.